dvc/dependency/param.py

```diff
@@ -54,14 +54,12 @@ def read_param_file(
                 continue
         return ret
 
-    from copy import deepcopy
-
     from dpath import merge
 
-    for key_path in key_paths:
+    for key_path in set(key_paths):
```

Comment:
Use [`funcy.distinct`](https://funcy.readthedocs.io/en/stable/seqs.html#distinct) here. Order is important.

```suggestion
    for key_path in ldistinct(key_paths):
```

Reply:
Thanks! 🤦🏼‍♂️ Resolved with https://github.com/iterative/dvc/pull/10180/commits/a3a8e45cc22e005e34afa8e600461d279ab82de8